ts/components/LabelledItem.tsx

```diff
@@ -74,144 +70,139 @@ type Props = WithTestID<CommonProp> &
-      this.props.inputProps.onChangeText(text);
+  const handleOnChangeText = (text: string) => {
+    if (props.type === "text" && props.inputProps.onChangeText) {
+      props.inputProps.onChangeText(text);
     }
-    this.checkInputIsEmpty(text);
+    checkInputIsEmpty(text);
   };
 
   /**
    * handle masked input on change text
    */
-  private handleOnMaskedChangeText = (formatted: string, text?: string) => {
-    if (
-      this.props.type === "masked" &&
-      this.props.inputMaskProps.onChangeText
-    ) {
-      this.props.inputMaskProps.onChangeText(formatted, text);
+  const handleOnMaskedChangeText = (formatted: string, text?: string) => {
+    if (props.type === "masked" && props.inputMaskProps.onChangeText) {
+      props.inputMaskProps.onChangeText(formatted, text);
     }
-    this.checkInputIsEmpty(text);
+    checkInputIsEmpty(text);
   };
 
   /**
    * keep track if input (or masked input) gains focus
    */
-  private handleOnFocus = () => {
-    this.setState({ hasFocus: true });
+  const handleOnFocus = () => {
+    setHasFocus(true);
   };
 
   /**
    * keep track if input (or masked input) loses focus
    */
-  private handleOnBlur = () => {
-    this.setState({ hasFocus: false });
+  const handleOnBlur = () => {
+    setHasFocus(false);
   };
 
-  public render() {
-    return (
-      <View>
+  const descriptionColor = props.isValid === false ? "red" : "bluegreyDark";
+  const accessibilityLabel = props.accessibilityLabel ?? "";
+  const inputBorderColor =
+    hasFocus && isEmpty
+      ? variables.itemBorderDefaultColor
+      : props.focusBorderColor;
+  const isValid = props.isValid === undefined ? false : props.isValid;
+  return (
+    <View>
+      <View
+        importantForAccessibility="no-hide-descendants"
+        accessibilityElementsHidden={true}
+      >
         <Item style={styles.noBottomLine}>
-          <H5
-            accessibilityRole="header"
-            accessibilityLabel={`${this.props.label}, ${I18n.t(
-              "global.accessibility.inputLabel"
-            )}`}
-          >
-            {this.props.label}
-          </H5>
+          <H5>{props.label}</H5>
         </Item>
+      </View>
+
+      <View
+        accessible={true}
+        accessibilityLabel={I18n.t("global.accessibility.textField", {
+          inputLabel: accessibilityLabel
+        })}
+        accessibilityHint={props.accessibilityHint}
+      >
         <Item
           style={{
             ...styles.bottomLine,
-            borderColor:
-              this.state.hasFocus && this.state.isEmpty
-                ? variables.itemBorderDefaultColor
-                : this.props.focusBorderColor
+            borderColor: inputBorderColor
           }}
-          error={this.props.isValid === undefined ? false : !this.props.isValid}
-          success={
-            this.props.isValid === undefined ? false : this.props.isValid
-          }
+          error={!isValid}
+          success={isValid}
```

Comment:
```typescript
const isError = props.isValid === undefined ? false : !props.isValid;
const isSuccess = props.isValid === undefined ? false : props.isValid;
```

```suggestion
          error={isError}
          success={isSuccess}
```
This is not an exact porting of the previous code and this is the effect

| before  | after |
| ------------- | ------------- |
![before](https://user-images.githubusercontent.com/822471/117999221-e9683200-b344-11eb-933d-cf3378c31785.png) | ![after](https://user-images.githubusercontent.com/822471/117999223-ea995f00-b344-11eb-8d7f-8572cdb847cd.png)

this is because these lines could have these values

### success (now) 
`const isValid = props.isValid === undefined ? false : props.isValid;`
### success (before)
`this.props.isValid === undefined ? false : this.props.isValid`

### success (no changes - it is the same)
| props.isValid  | before | now |
| ------------- | ------------- | ------------- |
| undefined  | false | false |
| true  | true | true |
| false  | false | false |

### error (now) 
(it is computed as !isValid)
### error (before)
`this.props.isValid === undefined ? false : !this.props.isValid`

### error 
| props.isValid  | before | now |
| ------------- | ------------- | ------------- |
| undefined  | false | true 🐞 |
| true  | false | false |
| false  | true | true |

Reply:
You're right! Thank you for catch it!
Fixed in [794a997](https://github.com/pagopa/io-app/pull/3020/commits/794a99712616180670cac96fbf8ef6294fe0af36)